tests/contrib/operators/test_kubernetes_job_operator.py

```diff
@@ -0,0 +1,116 @@
+                            '0 Running / 1 Succeeded / 0 Failed',
+                            'cleaned up']
+
+    FAILURE_SIDE_EFFECTS = ['applied',
+                            '1 Running / 0 Succeeded / 0 Failed',
+                            '0 Running / 0 Succeeded / 1 Failed',
+                            'cleaned up']
+
+    def setUp(self):
+        kubernetes_operator.SLEEP_TIME_BETWEEN_POLLING = 1
+
+    @mock.patch('subprocess.check_output')
+    def kubernetes_job_operator_execution(self,
+                                          success,
+                                          kubernetes_job_operator_task,
+                                          mock_check_output):
+        if success:
+            mock_check_output.side_effect = self.SUCCESS_SIDE_EFFECTS
+        else:
+            mock_check_output.side_effect = self.FAILURE_SIDE_EFFECTS
+
+        task_instance = TaskInstance(task=kubernetes_job_operator_task,
+                                     execution_date=datetime.utcnow() - timedelta(days=1))
+
+        if success:
+            task_instance.run()
+            self.assertEquals('success', task_instance.current_state())
+        else:
+            # Failed pod will cause task_instance to fail
+            with self.assertRaises(Exception):
+                task_instance.run()
+            self.assertEquals('failed', task_instance.current_state())
+
+    def test_basic_execution_failure(self):
+        task = kubernetes_operator.KubernetesJobOperator(kubernetes_job_name='test-basic',
+                                                         kubernetes_job_yaml_dictionary={
+                                                            'containers': [self.CONTAINER]
+                                                         },
+                                                         task_id=self.TASK_ID,
+                                                         dag=self.DAG)
+        self.kubernetes_job_operator_execution(success=False,
+                                               kubernetes_job_operator_task=task)
+
+    @mock.patch('subprocess.check_output')
+    def test_basic_execution_success(self, mock_check_output):
+        task = kubernetes_operator.KubernetesJobOperator(kubernetes_job_name='test-basic',
+                                                         kubernetes_job_yaml_dictionary={
+                                                            'containers': [self.CONTAINER]
+                                                         },
+                                                         task_id=self.TASK_ID,
+                                                         dag=self.DAG)
+        self.kubernetes_job_operator_execution(success=True,
+                                               kubernetes_job_operator_task=task)
+
+    @mock.patch('subprocess.check_output')
+    def test_user_provided_yaml_execution_failure(self, mock_check_output):
```

Comment:
these tests are pretty repetitive. Couldn't `kubernetes_job_operator_execution` just create the `KubernetesJobOperator` to avoid the duplication? It seems the only things that change are:

* kubernetes_job_yaml_dictionary
* kubernetes_job_yaml_template
* success

Maybe just pass those arguments into some helper, like `kubernetes_job_operator_execution` and have it do all the work?